CHANGELOG.md

```diff
@@ -6,6 +6,10 @@
 
 ### New Features and Enhancements
 
+The `N_VLinearSum` and `N_VLinearSumVectorArray` operations now assume that the output 
+can be used as input only in the first component, ensuring compatibility with external 
```

Comment:
This seems sort of like we are breaking the API, which would require a new major version. The reason I say "sort of" is because the existing vectors would still work since our change is in the assumption not the implementations. However, the change in the assumption now instructs us to create new vectors that would possibly not work with old code.

Reply:
I disagree, since nothing about this PR breaks any existing/working code.  

The only thing that would "break" is if a user creates an N_Vector that works with SUNDIALS v7.2.1, and then tries to use that same vector with an older version of SUNDIALS.  If anything, the documentation update could add a `.. versionchanged` comment to note that previous SUNDIALS versions required that all vectors support the case where either input could also be the output.

Reply:
>I disagree, since nothing about this PR breaks any existing/working code.

That why I said it was "sort of" a breaking change. It changes the contract of the API, therefore pedantically speaking it is a breaking change to the API. However, in practice, its impact is more like a "new feature".